src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1055,25 +1055,37 @@ void java_bytecode_parsert::rcode_attribute(methodt &method)
   }
   else if(attribute_name=="LocalVariableTypeTable")
   {
-    u2 local_variable_table_length=read_u2();
-
-    method.local_variable_table.resize(local_variable_table_length);
+    u2 local_variable_type_table_length=read_u2();
 
-    for(std::size_t i=0; i<local_variable_table_length; i++)
+    INVARIANT(
+      local_variable_type_table_length<=method.local_variable_table.size(),
+      "LVT must have at least as many elements as the type table.");
```

Comment:
I would expression the condition and the comment in the same way:

```
method.local_variable_table.size() >= local_variable_type_table_length
```

I would probably also use variable table instead of LVT to be consistent with type table (or change both to acronyms, as I initially assumed LVT=LVTT).